### PR TITLE
Add source and provider definition tooltips

### LINKED ISSUE #2775: Update the single result page
## Problem

Additional search views project updates the design of the single result page. This is a meta-issue for the single result page changes.

## Description

These changes should be conditional on whether the `additional_search_views` feature flag is enabled.

The Figma links for new designs: [Image single result](https://www.figma.com/file/niWnCgB7K0Y4e4mgxMrnRC/Additional-search-views?type=design&amp;node-id=1200-63284&amp;mode=dev) [Audio single result](https://www.figma.com/file/niWnCgB7K0Y4e4mgxMrnRC/Additional-search-views?type=design&amp;node-id=1200-63285&amp;mode=dev)

### Update the VCollectionLink area on the single result page
- [x] #2730
The content info line under the main item on the single result page should be replaced with a section that has two buttons: one for a creator link and a source link. This section should be horizontally scrollable on mobile. It should implement a scroll-snap (example: https://play.tailwindcss.com/AbfA33Za50)

### Use VTag with links for the tags
- [x] #2773
- [x] #2730
The tags should be rendered using the VTag component with links to the tag collection page.

### Update the tags area on the single result page
- [ ] #2589
The tags area should be collapsible to make long lists of tags collapsible.

### Add the information popover next to the source and provider links
- [ ] #2840
The information popover should be added next to the source and provider links that explains the difference between the source and provider - this is optional in this issue, can be moved to a separate one

## Mockups
[Figma link](https://www.figma.com/file/niWnCgB7K0Y4e4mgxMrnRC/Additional-search-views?type=design&amp;node-id=1200-56521&amp;mode=dev)

### LINKED ISSUE #2840: Source and provider definitions in single result page
## Problem

The Additional search views' designs (#2113) update the metadata section in single result page to distinguish between source and providers, but mockups did not include the tooltip texts triggered by the new Metadata element design (#2582).

## Description

I'd like your thoughts in the following copy for the two tooltips:

* **Source:** Organization granting the content
* **Provider:** Site hosting the content

In context, this looks like this

https://github.com/WordPress/openverse/assets/895819/f5df206e-9bc6-4353-85e1-d326c2e194e4

Tooltip texts should remain short and clear, so ideally we should keep this sentence as simple as possible. Acknowledging that [sources page](https://openverse.org/sources) gives more detail about these two concepts.

What do you think of these texts?

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #2775 by @fcoveram
Fixes #2840 by @fcoveram 

## Description

Adds the tooltip that opens when you focus or hover over the (i) symbol next to the Provider or Source in the single result page:
Screenshot 2023-12-18 at 2 43 16 PM

I just realized that the provider definition would not be correct when/if we extract sources from Europeana: here, Europeana is an aggregator which allows us to search for images, but the images themselves are hosted on the __source__ sites, not Europeana.

### The tooltip
Here are the Figma links to: 
* [Tooltip component](https://www.figma.com/file/GIIQ4sDbaToCfFQyKMvzr8/Openverse-Design-Library?type=design&node-id=5354-29979&mode=dev)
* [Help component](https://www.figma.com/file/GIIQ4sDbaToCfFQyKMvzr8/Openverse-Design-Library?type=design&node-id=2484-22552&mode=dev). Tooltip is visible in `hover` state

### Accessibility considerations
[WAI ARIA guidelines](https://www.w3.org/TR/wai-aria-1.1/#tooltip)

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [ ] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by AetherUnbound — APPROVED** [review]
Looked over the code and it (mostly) makes sense to me 😄 tool tips show up great when testing locally!

File: frontend/src/components/VMediaInfo/VMetadata.vue [hunk 1]
```diff
@@ -1,7 +1,15 @@
...
+    <div v-for="datum in metadata" :key="datum.label">
+      <VSourceProviderTooltip
+        v-if="tooltipId(datum)"
+        :describedby="tooltipId(datum)"
+        class="label-regular mb-1 flex flex-row ps-1"
+        :datum="datum"
+      />
+      <dt v-else class="label-regular mb-1 flex flex-row ps-1">
+        {{ $t(datum.label) }}
+      </dt>
```
Comment (fcoveram) [inline frontend/src/components/VMediaInfo/VMetadata.vue:12]: The info icon is not horizontally centered. I tried the following in Chrome inspector, and it looked good:
* Added `items-center`
* Just in `xs`, I added a padding of 4px to top, right, and bottom in `dt` element. For bigger breakpoints, `ps-1` works correctly

**Review by fcoveram — DISMISSED** [review]
I noticed two minor details that polish the layout. All the rest look great 👌

File: frontend/src/components/VMediaInfo/VMetadata.vue (hunk 1 above)
Comment (obulat) [reply frontend/src/components/VMediaInfo/VMetadata.vue:12]: Sorry, @fcoveram , I didn't mean to re-request your review yet

Comment (fcoveram) [reply frontend/src/components/VMediaInfo/VMetadata.vue:12]: No problem!

File: frontend/src/components/VMediaInfo/VSourceProviderTooltip.vue [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    const i18n = useI18n()
+    const description = computed(() => {
+      if (!props.datum.name) {
+        return ""
+      }
+      return i18n.t(
+        props.datum.name === "source"
+          ? "mediaDetails.sourceDescription"
+          : "mediaDetails.providerDescription"
+      )
```
Comment (dhruvkb) [inline frontend/src/components/VMediaInfo/VSourceProviderTooltip.vue:46]:
```suggestion
      return i18n.t(`mediaDetails.${props.datum.name}Description`)
```

File: frontend/src/components/VMediaInfo/VSourceProviderTooltip.vue [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+export default defineComponent({
+  name: "VSourceProviderTooltip",
+  components: { VTooltip },
+  props: {
+    datum: {
+      type: Object as PropType<Metadata>,
+      required: true,
+    },
+    describedby: {
```
Comment (dhruvkb) [inline frontend/src/components/VMediaInfo/VSourceProviderTooltip.vue:31]: Nit: it's bugging me that this does not follow camel case for the 'B' as in `describedBy`.

**Review by dhruvkb — APPROVED** [review]
LGTM, except to small notes. Nothing to block!

File: frontend/src/components/VMediaInfo/VSourceProviderTooltip.vue (hunk 2 above)
Comment (obulat) [reply frontend/src/components/VMediaInfo/VSourceProviderTooltip.vue:46]: Static keys can be checked by [`vue-i18n-extract`](https://github.com/Spittal/vue-i18n-extract)`. It does not handle dynamic keys like this well.

**Pull request merged**